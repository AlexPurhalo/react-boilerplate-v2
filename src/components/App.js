import React       from 'react'
import gql         from 'graphql-tag'
import { graphql } from 'react-apollo'

const GET_TODOS = gql`
query {
  allItems {
    edges {
      node {
        id
        body
      }
    }
  }
}
`

const App = ({ data }) => {
	const { loading } = data
	
	if (loading) return <div>...loading</div>
	
	const { allItems: { edges: items } } = data
	
	return (
		<ul>
			{items.map(({ node: { id, body } }) =>
				<li key={id}>{body}</li>
			)}
		</ul>
	)
}

export default graphql(GET_TODOS)(App)
