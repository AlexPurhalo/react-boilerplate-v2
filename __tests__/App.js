import React                  from 'react'
import { configure, shallow } from 'enzyme'
import Adapter                from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() })

import App from '../src/components/App'

// Snapshot test
test('renders a main component', () => {
	const component = renderer.create(<App/>);
	
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

// Test via enzyme usage
test('has the right text', () => {
	const component = shallow(<App />)
	expect(component.text()).toEqual('Hello and welcome :)');
});