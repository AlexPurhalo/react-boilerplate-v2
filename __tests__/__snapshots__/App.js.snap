// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a main component 1`] = `
<h1>
  Hello and welcome :)
</h1>
`;
